src/storage/mvcc/reader/backward_scanner.rs

```diff
@@ -0,0 +1,758 @@
+                // `REVERSE_SEEK_BOUND - 1` times.
+                self.write_cursor.prev(&mut self.statistics.write);
+                if !self.write_cursor.valid() {
+                    // Key space ended. We use `last_version` as the return.
+                    return Ok(self.handle_last_version(last_version, user_key)?);
+                }
+            }
+
+            let mut is_done = false;
+            {
+                let current_key = self.write_cursor.key(&mut self.statistics.write);
+                last_checked_commit_ts = Key::decode_ts_from(current_key)?;
+
+                if !Key::is_user_key_eq(current_key, user_key.encoded().as_slice())
+                    || last_checked_commit_ts > ts
+                {
+                    // Meet another key or meet an unwanted version. We use `last_version` as the return.
+                    // TODO: If we meet another user key here, we don't need to compare the key
+                    // again when trying to move to prev user key in `read_next`.
+                    is_done = true;
+                }
+            }
+            if is_done {
+                return Ok(self.handle_last_version(last_version, user_key)?);
+            }
+
+            let write = Write::parse(self.write_cursor.value(&mut self.statistics.write))?;
+            self.statistics.write.processed += 1;
+
+            match write.write_type {
+                WriteType::Put | WriteType::Delete => last_version = Some(write),
+                WriteType::Lock | WriteType::Rollback => {}
+            }
+        }
+
+        // At this time, we must have current commit_ts <= ts. If commit_ts == ts,
+        // we don't need to seek any more and we can just utilize `last_version`.
+        if last_checked_commit_ts == ts {
+            return Ok(self.handle_last_version(last_version, user_key)?);
+        }
+        assert!(ts > last_checked_commit_ts);
+
+        // After several `prev()`, we still not get the latest version for the specified ts,
+        // use seek to locate the latest version.
+        // `user_key` must have reserved space here, so its clone has reserved space too. So no
+        // reallocation happends in `append_ts`.
+        let seek_key = user_key.clone().append_ts(ts);
+
+        // TODO: Replace by cast + seek().
+        self.write_cursor
+            .internal_seek(&seek_key, &mut self.statistics.write)?;
+        assert!(self.write_cursor.valid());
+
+        loop {
+            // After seek, or after some `next()`, we may reach `last_checked_commit_ts` again. It
+            // means we have checked all versions for this user key. We use `last_version` as
+            // return.
+            let mut is_done = false;
+            {
+                let current_key = self.write_cursor.key(&mut self.statistics.write);
+                // We should never reach another user key.
+                assert!(Key::is_user_key_eq(
```

Comment:
They are must be the same row, so we can skip the step that checking tail 8 bytes?